### PR TITLE
feat(7175): Addition of KubesliceEvents in Sliceconfig Status | CRD updates

### PR DESCRIPTION
**Pull request by priyank-upadhyay**
# Description
Added `KubesliceEvents` in `SliceConfig > Status`: 
```go
// SliceConfigStatus defines the observed state of SliceConfig
type SliceConfigStatus struct {
	KubesliceEvents []KubesliceEvent `json:"kubesliceEvents,omitempty"`
}

type KubesliceEvent struct {
	// Type of the event. Can be one of Error, Success or InProgress
	Type string `json:"type,omitempty"`
	// Trigger action. Examples - CLUSTER_OFFBOARDING, NAMESPCE_OFFBOARDING etc
	Action string `json:"action,omitempty"`
	// list of effected components on which action failed
	Components []string `json:"components,omitempty"`
	// Identifier of the component for which the action was triggered
	Identifier string `json:"identifier,omitempty"`
	// Reason message for the event
	Reason string `json:"reason,omitempty"`
	// Event name (from monitoring framework schema)
	Event string `json:"event"`
	// Timestamp of the event
	Timestamp metav1.Time `json:"timestamp,omitempty"`
	// Flag to determine if kubernetes event is already raised
	//+kubebuilder:default:=false
	IsEventRaised bool `json:"isEventRaised,omitempty"`
}
```

Fixes [AM-7175](https://avesha.atlassian.net/browse/AM-7175)

## How Has This Been Tested?

- [x] Manual testing

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [x] I have tested it for all user roles.
* [ ] I have added all the required unit test cases.

## Does this PR introduce a breaking change for other components like worker-operator?

```release-note

```

[AM-7175]: https://avesha.atlassian.net/browse/AM-7175?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apis/controller/v1alpha1/sliceconfig_types.go [hunk 1]
```diff
@@ -106,8 +106,26 @@ type SliceNamespaceSelection struct {
 	Clusters  []string `json:"clusters,omitempty"`
 }
 
+type KubesliceEvent struct {
```
Comment (PuranAdhikari) [inline apis/controller/v1alpha1/sliceconfig_types.go:109]: Do we actually need this? Without this as well, when we do `kubectl describe sliceconfig <name>`, we will get all the events associated to that slice config within the CR.

Comment (priyank-upadhyay) [reply apis/controller/v1alpha1/sliceconfig_types.go:109]: Kubernetes events are transient in nature. we want to persist the events from some of the action in this use case. 
Moreover these events would be coming from API-GW which does not have any framework to implement to ensure consistency in event generation.

Comment (PuranAdhikari) [reply apis/controller/v1alpha1/sliceconfig_types.go:109]: Cool. That make sense.

**Review by PuranAdhikari — APPROVED** [review]

File: apis/controller/v1alpha1/sliceconfig_types.go [hunk 2]
```diff
@@ -106,8 +106,26 @@ type SliceNamespaceSelection struct {
...
+	Reason string `json:"reason,omitempty"`
+	// Event name (from monitoring framework schema)
+	Event string `json:"event"`
+	// Timestamp of the event
+	Timestamp metav1.Time `json:"timestamp,omitempty"`
+}
+
 // SliceConfigStatus defines the observed state of SliceConfig
 type SliceConfigStatus struct {
+	KubesliceEvents []KubesliceEvent `json:"kubeslicEvent,omitempty"`
```
Comment (PuranAdhikari) [inline apis/controller/v1alpha1/sliceconfig_types.go:128]: Small typo in `kubeslicEvent` -> `kubesliceEvents`

**Review by PuranAdhikari — CHANGES_REQUESTED** [review]

**Review by PuranAdhikari — APPROVED** [review]
LGTM.

**Comment by priyank-upadhyay** [comment]
Screenshot 2023-03-29 at 11 45 28 AM

**Pull request merged**